pkg/internal/itest/http_fetch_test.go

```diff
@@ -990,13 +990,18 @@ func TestHttpFetch(t *testing.T) {
 			}
 
 			for i, resp := range responses {
-				if testCase.expectFail {
-					req.Equal(http.StatusGatewayTimeout, resp.StatusCode)
+				if testCase.expectNoCandidates {
+					if resp.StatusCode != http.StatusBadGateway {
```

Comment:
This is a notable change in the testing, we go from expecting a 504 here to a 502, with an explicit "no candidates found". Previously there was no error because the bitswap retriever, uniquely, did a `return nil, nil` (stats, error). Now that's not an option, it always returns a result or an error. Now in this test case it'll return an `ErrNoCandidates`, because the async candidate finder returns with no candidates, which maps to 502. In practice  it's going to depend on the http request to the indexer timing our or returning early, so we may get a timeout.